src/main/java/frc/robot/maps/FrancoisMap.java

```diff
@@ -98,42 +103,93 @@ public ControlPanelMap getControlPanelMap() {
+            AnalogTrigger backIntakeIR = new AnalogTrigger(2);
+            AnalogTrigger frontIntakeIR = new AnalogTrigger(3);
+
+            @Override
+            public SendableSpeedController pierreMotor() {
+                final WPI_TalonSRX pierreMotor = new WPI_TalonSRX(40);
+                return SendableSpeedController.wrap(pierreMotor);
+            }
+
+            public SendableSpeedController singulator() {
+                final WPI_TalonSRX singulator = new WPI_TalonSRX(41);
+                singulator.setInverted(true);
+                return SendableSpeedController.wrap(singulator);
+            }
+
+            public BooleanSupplier topPierreIR() {
+                topPierreIR.setLimitsVoltage(1.2, 1.4);
+                return topPierreIR::getTriggerState;
+            }
+
+            public BooleanSupplier bottomPierreIR() {
+                bottomPierreIR.setLimitsVoltage(1.2, 1.4);
+                return bottomPierreIR::getTriggerState;
+            }
+
+            public BooleanSupplier backIntakeIR() {
+                backIntakeIR.setLimitsVoltage(1.2, 2.6);
+                return backIntakeIR::getTriggerState;
+            }
+
+            public BooleanSupplier frontIntakeIR() {
+                frontIntakeIR.setLimitsVoltage(1.2, 1.4);
+                return frontIntakeIR::getTriggerState;
+            }
+
+        };
+    }
+
     @Override
     public LiftMap getLiftMap() {
         return new LiftMap() {
             CANSparkMax follower = new CANSparkMax(21, MotorType.kBrushless);
             CANSparkMax leader = new CANSparkMax(28, MotorType.kBrushless);
             InvertDigitalInput upperLimit = new InvertDigitalInput(0);
+            InvertDigitalInput lowerLimit = new InvertDigitalInput(1);
 
             @Override
             public PIDSparkMax elevator() {
-                follower.follow(leader);
+                leader.setInverted(true);
+                follower.follow(leader, true);
+                leader.setIdleMode(IdleMode.kBrake);
+                follower.setIdleMode(IdleMode.kBrake);
 
                 return new PIDSparkMax(leader);
             }
 
             @Override
             public ISolenoid liftBrake() {
-                // TODO Change this to a real Channel
-                WSolenoid brake = new WSolenoid(3);
-
+                WSolenoid brake = new WSolenoid(0);
                 return brake;
             }
 
             @Override
             public BooleanSupplier upperLiftLimit() {
+                upperLimit.setInverted(true);
                 return upperLimit::get;
             }
 
+            @Override
+            public BooleanSupplier lowerLiftLimit() {
+                return lowerLimit::get;
+            }
+
             @Override
             public IEncoder getLiftEncoder() {
                 SparkMaxEncoder getEncoder = new SparkMaxEncoder(leader.getEncoder());
-
+                getEncoder.setScaleFactor(1);
```

Comment:
Lets figure out the actual scale factor here.

Reply:
Fixed